File: app/src/main/java/im/point/dotty/mapper/PostMapper.java
package im.point.dotty.mapper;

import java.text.ParseException;
import java.text.SimpleDateFormat;

import im.point.dotty.model.Post;
import im.point.dotty.network.MetaPost;

class PostMapper {
    protected final static SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");

    protected static <T extends Post> T map(T result, MetaPost post) {
        try {
            result.id = post.uid;
            result.textId = post.post.id;
            result.userId = post.post.author.id;
            result.login = post.post.author.login;
            result.name = post.post.author.name;
            result.text = post.post.text;
            result.timestamp = format.parse(post.post.created);
        } catch (ParseException e) {
            return null;
        }
        return result;
    }
}

File: app/src/main/java/im/point/dotty/mapper/PostMapper.kt
/*
 * Copyright (c) 2019-2021 Pavel A. Sokolov
 */
package im.point.dotty.mapper

import im.point.dotty.model.CompletePost
import im.point.dotty.model.Post
import im.point.dotty.network.RawPost

class PostMapper : MetaPostMapper<CompletePost<*>>(), Mapper<Post, RawPost> {
    override fun map(entry: RawPost): Post {
        with(entry) {
            val id = id ?: throw Exception("Post ID is not provided")
            val authorId = author?.id ?: throw Exception("Post's author ID is not provided")
            val login = author?.login ?: throw Exception("Post's author login is not provided")
            return mergeRawPost(Post(id, authorId, login), entry)
        }
    }
}